### PR TITLE
Fix SM (and some other) follow links

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Sets the follow target of the SM crystal's internal headset (yes, that is how indeed it works) to the crystal itself, andalso updates this for some similar implementations.

## Why It's Good For The Game
Should make clicking "follow" on the SM announcements do what you'd expect it to

## Testing
set off a delam, followed it

## Changelog
:cl:
fix: The follow link on SM crystal (and some other) announcements should work more consistently. 
/:cl:

### CODE REVIEW

File: code/game/machinery/cloning.dm [hunk 1]
```diff
@@ -70,6 +70,7 @@ GLOBAL_LIST_INIT(cloner_biomass_items, list(\
 	Radio = new /obj/item/radio(src)
 	Radio.listening = FALSE
 	Radio.config(list("Medical" = 0))
+	radio.follow_target = src
```
Comment (Contrabang) [inline code/game/machinery/cloning.dm:73]:
```suggestion
	Radio.follow_target = src
```

File: code/modules/projectiles/guns/energy/special_eguns.dm [hunk 2]
```diff
@@ -754,6 +754,7 @@
 	. = ..()
 	Announcer = new /obj/item/radio/headset(src)
 	Announcer.config(list("Security" = 1))
+	announcer.follow_target = src
```
Comment (Contrabang) [inline code/modules/projectiles/guns/energy/special_eguns.dm:757]:
```suggestion
	Announcer.follow_target = src
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
minor capitalization changes

**Review by S34NW — APPROVED** [review]
Good man, is this also possible for the arrivals announcer?

**Comment by lewcc** [comment]
It certainly is possible, implemented.

File: code/modules/mob/living/silicon/ai/latejoin.dm [hunk 3]
```diff
@@ -13,7 +13,7 @@ GLOBAL_LIST_EMPTY(empty_playable_ai_cores)
 
 /mob/living/silicon/ai/proc/cryo_AI()
 	GLOB.empty_playable_ai_cores += new /obj/structure/AIcore/deactivated(loc)
-	GLOB.global_announcer.autosay("[src] has been moved to intelligence storage.", "Artificial Intelligence Oversight")
+	GLOB.global_announcer.autosay("[src] has been moved to intelligence storage.", "Artificial Intelligence Oversight", follow_target_override = src)
```
Comment (Contrabang) [inline code/modules/mob/living/silicon/ai/latejoin.dm:16]: I think this should follow the new AIcore/deactivated since src gets deleted later in this proc

**Review by AffectedArc07 — APPROVED** [review]

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
What contra said

File: code/modules/mob/living/silicon/ai/latejoin.dm (hunk 3 above)
Comment (lewcc) [reply code/modules/mob/living/silicon/ai/latejoin.dm]: Addressed, good catch!

**Comment by lewcc** [comment]
Changes addressed

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**